### PR TITLE
Update target version to 12.0

### PR DESCRIPTION
**Pull request by MikhailSuendukov**
* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

During the build process, warnings indicate that AppCenter targets iOS 11, even though the minimum supported version is iOS 12. So we raised target versions for iOS and tvOS to 12.0.
image

## Related PRs or issues
[AB#107365](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/107365)

### CODE REVIEW

**Comment by DmitriyKirakosyan** [comment]
Update SPM

File: Package.swift [hunk 1]
```diff
@@ -54,10 +54,10 @@ let cHeaderSearchPaths: [CSetting] = projectHeaderSearchPaths.map { .headerSearc
 let package = Package(
     name: "AppCenter",
     platforms: [
-        .iOS(.v11),
+        .iOS(.v12),
```
Comment (DmitriyKirakosyan) [inline Package.swift:57]: no need to change this file, it's for older XCode. It's probably a good time to remove this file.

Comment (MikhailSuendukov) [reply Package.swift]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## Version 5.0.6 (Under development)
 
 * **[Improvement]** Update target iOS and tvOS version to 12.0.
+* **[Improvement]** Bump minimum Xcode version to 12.
```
Comment (DmitriyKirakosyan) [inline CHANGELOG.md:6]: Remove this line. Currently we support Xcode 13 and higher.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ The goal of this document is to provide a high-level overview of how you can get
 
 To contribute to the SDK, please
 
-* Install Xcode 11 on your Mac.
+* Install Xcode 12 on your Mac.
```
Comment (DmitriyKirakosyan) [inline CONTRIBUTING.md:8]:
```suggestion
* Install Xcode 13 on your Mac.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ Please make sure the following is done when submitting a pull request:
 
 1. Fork the repository and create your branch from `develop`.
 1. Run `git submodule update --init --recursive` before opening the solution.
-1. Use Xcode 11 or above to edit and compile the SDK.
+1. Use Xcode 12 or above to edit and compile the SDK.
```
Comment (DmitriyKirakosyan) [inline CONTRIBUTING.md:22]:
```suggestion
1. Use Xcode 13 or above to edit and compile the SDK.
```

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**